lib/block_view_dao_coin_limit_order.go

```diff
@@ -1062,16 +1066,147 @@ func (bav *UtxoView) _getDAOCoinLimitOrderEntry(inputEntry *DAOCoinLimitOrderEnt
 	return dbAdapter.GetDAOCoinLimitOrder(inputEntry)
 }
 
+func (bav *UtxoView) _getAllDAOCoinLimitOrdersForThisDAOCoinPair(
+	buyingDAOCoinCreatorPKID *PKID, sellingDAOCoinCreatorPKID *PKID) ([]*DAOCoinLimitOrderEntry, error) {
+	// This function is used by the API to construct all open
+	// orders for the input buying and selling DAO coins.
+	if buyingDAOCoinCreatorPKID == nil {
+		return nil, errors.Errorf("_getAllDAOCoinLimitOrdersForThisDAOCoinPair: Called with nil buy coin PKID; this should never happen")
+	}
+	if sellingDAOCoinCreatorPKID == nil {
+		return nil, errors.Errorf("_getAllDAOCoinLimitOrdersForThisDAOCoinPair: Called with nil sell coin PKID; this should never happen")
+	}
+
+	outputEntries := []*DAOCoinLimitOrderEntry{}
+
+	// We want to only return a unique list of orders. We use this map as
+	// a Set to dedup orders between the UTXO view and the database with
+	// preference given to orders from the UTXO view as they are newer.
+	seenOrderEntries := map[string]bool{}
+
+	// Get orders from the UTXO view.
+	for _, orderEntry := range bav.DAOCoinLimitOrderMapKeyToDAOCoinLimitOrderEntry {
+		if orderEntry.BuyingDAOCoinCreatorPKID.Eq(buyingDAOCoinCreatorPKID) &&
+			orderEntry.SellingDAOCoinCreatorPKID.Eq(sellingDAOCoinCreatorPKID) {
+			// Note: we can't just take ToMapKey() here as that does
+			// not include the quantity or block height in the struct.
+			orderEntryBytes, err := orderEntry.ToBytes()
+			if err != nil {
+				// This should never happen as DAO coin limit
+				// order ToBytes() never returns an error.
+				return nil, err
+			}
+			orderEntryString := string(orderEntryBytes)
+
+			outputEntries = append(outputEntries, orderEntry)
+			seenOrderEntries[orderEntryString] = true
+		}
+	}
+
+	// Get orders from the database.
+	dbAdapter := DbAdapter{
```

Comment:
should we add an function on UtxoView called `GetDBAdapter` so we don't have this construction going on everywhere?